app/services/ClaimToAdjustService.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import auth.MtdItUser
+import connectors.FinancialDetailsConnector
+import models.financialDetails.{DocumentDetail, FinancialDetailsModel}
+import models.incomeSourceDetails.TaxYear
+import models.incomeSourceDetails.TaxYear.makeTaxYearWithEndYear
+import play.api.Logger
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ClaimToAdjustService @Inject()(val financialDetailsConnector: FinancialDetailsConnector,
+                                     val financialDetailsService: FinancialDetailsService)
+                                    (implicit ec: ExecutionContext) {
+
+  def getPoATaxYear(implicit hc: HeaderCarrier, user: MtdItUser[_]): Future[Either[Throwable, Option[TaxYear]]] = {
+    val a = financialDetailsService.getAllFinancialDetails map {
```

Comment:
this might be not a good idea to create dependency on other service, like financialDetailsService;
if we can rely on the financialDetailsConnector;

Reply:
getAllFinancialDetails does pretty much everything we want though. I could duplicate the code in this service though..?

Reply:
you can duplicate this code in this service, but if you would look at the implementation, we should not do so many API calls;